src-tauri/src/domain/impls/tr17/section_2/equation_1.rs

```diff
@@ -0,0 +1,275 @@
+use crate::domain::method::{step::Step, Method};
+use std::default;
+use std::sync::{Arc, RwLock};
+use std::vec;
+
+use super::super::super::Document;
+use super::super::TR17Section;
+use super::Section2Method;
+
+pub struct TR17Section2Equation1Builder;
+
+impl MethodBuilderTrait for TR17Section2Equation1Builder {
+    fn name() -> String {
+        "Non-dimensional Heat Release Rate".to_string()
+    }
+    fn tags() -> Vec<Tag> {
+        vec![Tag::HRR]
+    }
+    fn description() -> Option<String> {
+        Some("Calulates the non-dimensional Heat Release Rate".to_string())
+    }
+    fn quick_calc_compatible() -> bool {
+        true
+    }
+    fn reference() -> Reference {
+        Reference(Document::TR17(Some(TR17Section::Two(
+            Section2Method::EquationOne,
+        ))))
+    }
+
+    fn form(params: &Parameters) -> crate::domain::method::form::Form {
+        let mut step_1 = FormStep::new(
+            "Input | Eq. 1",
+            "Input required to calculate the non-dimensional heat release rate.",
+        );
+        for param in params.values().into_iter() {
+            if param.id() == "Q*" {
+                continue;
+            }
+            step_1.add_field(param.to_field())
+        }
+        let factor = params.get_parameter("Q*");
+        step_1.add_intro();
+        step_1.add_equation(
+            factor
+                .read()
+                .unwrap()
+                .expression()
+                .as_ref()
+                .unwrap()
+                .generate_with_symbols()[0][0]
+                .clone(),
+        );
+        Form {
+            steps: vec![step_1],
+        }
+    }
+    fn parameters() -> Parameters {
+        let mut params = Parameters::new();
+
+        let q_dot = ParamBuilder::float("Q")
+            .name("Heat Release Rate")
+            .units("kW")
+            .min(0.0)
+            .max(1000000.0)
+            .required()
+            .build();
+
+        let rho_a = ParamBuilder::float("\\rho_a")
+            .name("Density of air at ambient temperature")
+            .units("kg/m^{3}")
+            .min(0.0)
+            .max(100.0)
+            .required()
+            .default_value(Some(ParameterValue::Float(1.2)))
+            .build();
+
+        let c_p = ParamBuilder::float("c_p")
+            .name("Heat capacity of air at ambient temperature")
+            .units("kJ/kgK")
+            .min(0.0)
+            .max(100.0)
+            .required()
+            .default_value(Some(ParameterValue::Float(1.0)))
+            .build();
+
+        let t_a = ParamBuilder::float("T_a")
+            .name("Ambient temperature")
+            .units("K")
+            .min(0.0)
+            .max(1000.0)
+            .required()
+            .default_value(Some(ParameterValue::Float(293.0)))
+            .build();
+
+        let g = ParamBuilder::float("g")
+            .name("Gravity acceleration")
+            .units("m/s^{2}")
+            .min(0.0)
```

Comment:
```suggestion
            .min_exclusive(0.0)
```

I should probably create a new param type called constant.

I'll put it on my to do list